src/main/scala/it/pagopa/interop/attributeregistryprocess/api/impl/AttributeRegistryApiServiceImpl.scala

```diff
@@ -140,40 +141,35 @@ final case class AttributeRegistryApiServiceImpl(
       } yield ()
 
       onComplete(result) {
-        loadCertifiedAttributesResponse[Unit](operationLabel)(_ => loadCertifiedAttributes200)
+        loadCertifiedAttributesResponse[Unit](operationLabel)(_ => loadCertifiedAttributes204)
       }
     }
 
-  def addNewAttributes(
-    attributeSeed: Seq[AttributeSeed]
-  )(implicit contexts: Seq[(String, String)]): Future[Set[Attribute]] = {
-
-    case class DeltaAttributes(attributes: Set[Attribute], seeds: Set[AttributeSeed]) {
-      def addAttribute(attr: Attribute): DeltaAttributes = copy(attributes = attributes + attr)
-
-      def addSeed(seed: AttributeSeed): DeltaAttributes = copy(seeds = seeds + seed)
-    }
+  private def addNewAttributes(
+    attributesSeeds: Seq[AttributeSeed]
+  )(implicit contexts: Seq[(String, String)]): Future[Unit] = {
 
     // calculating the delta of attributes
-    def delta(attrs: List[Attribute]): DeltaAttributes =
-      attributeSeed.foldLeft[DeltaAttributes](DeltaAttributes(Set.empty, Set.empty))((delta, seed) =>
+    def delta(attrs: Seq[PersistentAttribute]): Set[AttributeSeed] =
+      attributesSeeds.foldLeft[Set[AttributeSeed]](Set.empty)((delta, seed) =>
```

Comment:
please use a different name for the parameter, this is the same used for the function and can be confusing